utilities/pipelines/sharedScripts/Set-EnvironmentOnAgent.ps1

```diff
@@ -2,38 +2,44 @@
 
 #region Helper Functions
 <#
-    .SYNOPSIS
-    Installes given PowerShell modules
+.SYNOPSIS
+Installes given PowerShell modules
 
-    .DESCRIPTION
-    Installes given PowerShell modules
+.DESCRIPTION
+Installes given PowerShell modules
 
-    .PARAMETER Module
-    Modules to be installed, must be Object
-    @{
-        Name = 'Name'
-        Version = '1.0.0' # Optional
-    }
+.PARAMETER Module
+Required. Modules to be installed, must be Object
+@{
+    Name = 'Name'
+    Version = '1.0.0' # Optional
+}
 
-    .EXAMPLE
-    Install-CustomModule @{ Name = 'Pester' } C:\Modules
+.PARAMETER installedModules
+Optional. Modules that are already installed on the machine. Can be fetched via 'Get-Module -ListAvailable'
 
-    Installes pester and saves it to C:\Modules
+.EXAMPLE
+Install-CustomModule @{ Name = 'Pester' } C:\Modules
+
+Installes pester and saves it to C:\Modules
 #>
 function Install-CustomModule {
 
     [CmdletBinding(SupportsShouldProcess)]
     Param (
         [Parameter(Mandatory = $true)]
-        [Hashtable] $Module
+        [Hashtable] $Module,
+
+        [Parameter(Mandatory = $false)]
+        [object[]] $installedModules = @()
```

Comment:
To echo our MS PowerShell guidelines: [Use Pascal Case for Parameter Names](https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/strongly-encouraged-development-guidelines?view=powershell-7.2#use-pascal-case-for-parameter-names)
 and [Use Singular Parameter Names](https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/strongly-encouraged-development-guidelines?view=powershell-7.2#use-singular-parameter-names)

```suggestion
        [object[]] $InstalledModule = @()
```